include/xtensor/xmime.hpp

```diff
@@ -112,11 +112,22 @@ namespace xt
                           const std::size_t& edgeitems, const std::vector<I>& idx)
     {
         const auto& dim = expr.shape()[expr.dimension() - 2];
+        const auto& last_dim = expr.shape()[expr.dimension() - 1];
         std::string idx_str;
         std::for_each(idx.cbegin(), idx.cend(), [&idx_str](const auto& i) {
             idx_str += std::to_string(i) + ", ";
         });
 
+        std::size_t nb_ellipsis = 2 * edgeitems + 1;
+        if (last_dim <= 2 * edgeitems + 1)
+        {
+            nb_ellipsis = last_dim;
+        }
+        else
+        {
+            nb_ellipsis = 2 * edgeitems + 1;
```

Comment:
Can we leave out this case: it's the default value, e.g.
```cpp

        std::size_t nb_ellipsis = 2 * edgeitems + 1;
        if (last_dim <= nb_ellipsis)
        {
            nb_ellipsis = last_dim;
        }

```

Reply:
Sure :) I should have checked my code twice

Reply:
I do this *all* the time ;)